### PR TITLE
Clear cache before priming inventory

### PR DESCRIPTION
**Pull request by RekhaAparna01**
This commit adds code to clear cache of a FRU  from D-bus before publishing the primed data for the FRU which is not present.

### CODE REVIEW

File: include/utility/vpd_specific_utility.hpp [hunk 1]
```diff
@@ -466,5 +466,88 @@ inline bool findCcinInVpd(const nlohmann::json& i_JsonObject,
     logging::logMessage("VPD type not supported. Can't find CCIN");
     return false;
 }
+
+/**
+ * @brief API to reset data of the FRU on DBus
```
Comment (SunnySrivastava1984) [inline include/utility/vpd_specific_utility.hpp:471]: > the

nit "a", also period at the end.

File: include/utility/vpd_specific_utility.hpp [hunk 2]
```diff
@@ -466,5 +466,88 @@ inline bool findCcinInVpd(const nlohmann::json& i_JsonObject,
...
+        {
+            if (l_service.compare(constants::pimServiceName) !=
+                constants::STR_CMP_SUCCESS)
+            {
+                continue;
+            }
+
+            for (const auto& l_interface : l_interfaceList)
+            {
+                if ((l_interface.find("com.ibm.ipzvpd") != std::string::npos) ||
```
Comment (SunnySrivastava1984) [inline include/utility/vpd_specific_utility.hpp:501]: can we use constants::ipzVpdInf as it is already defined.

File: include/utility/dbus_utility.hpp [hunk 3]
```diff
@@ -58,6 +58,51 @@ inline types::MapperGetObject getObjectMap(const std::string& objectPath,
     return getObjectMap;
 }
 
+/**
+ * @brief An API to get property map for an interface.
+ *
+ * This API returns a map of property and its value of a particular
```
Comment (SunnySrivastava1984) [inline include/utility/dbus_utility.hpp:64]: nit: 
> of a particular


with respect to a particular interface.

File: src/worker.cpp [hunk 4]
```diff
@@ -742,6 +742,9 @@ bool Worker::primeInventory(const std::string& i_vpdFilePath)
             continue;
         }
 
+        vpdSpecificUtility::resetDataOnDbus(std::string(l_Fru["inventoryPath"]),
```
Comment (SunnySrivastava1984) [inline src/worker.cpp:745]: Add a comment stating that clearing data from Dbus if already exists.

**Review by SunnySrivastava1984 — CHANGES_REQUESTED** [review]

File: include/utility/dbus_utility.hpp (hunk 3 above)
Comment (RekhaAparna01) [reply include/utility/dbus_utility.hpp]: Done

File: include/utility/vpd_specific_utility.hpp (hunk 1 above)
Comment (RekhaAparna01) [reply include/utility/vpd_specific_utility.hpp]: Done

File: include/utility/vpd_specific_utility.hpp (hunk 2 above)
Comment (RekhaAparna01) [reply include/utility/vpd_specific_utility.hpp]: Done

File: src/worker.cpp (hunk 4 above)
Comment (RekhaAparna01) [reply src/worker.cpp:746]: Done

**Review by SunnySrivastava1984 — APPROVED** [review]

File: include/utility/vpd_specific_utility.hpp [hunk 5]
```diff
@@ -466,5 +466,89 @@ inline bool findCcinInVpd(const nlohmann::json& i_JsonObject,
     logging::logMessage("VPD type not supported. Can't find CCIN");
     return false;
 }
+
+/**
+ * @brief API to reset data of a FRU on DBus.
```
Comment (SunnySrivastava1984) [inline include/utility/vpd_specific_utility.hpp:471]: Nit: It only resets the data under PIM, Can you add that to the Doxygen?
Or you can rename the API as well
resetDataUnderPim?

**Review by SunnySrivastava1984 — CHANGES_REQUESTED** [review]

File: include/utility/vpd_specific_utility.hpp (hunk 5 above)
Comment (RekhaAparna01) [reply include/utility/vpd_specific_utility.hpp:471]: Done
Updated the API name

**Review by SunnySrivastava1984 — APPROVED** [review]

**Pull request merged**